### PR TITLE
Feat: apply-component supports namespace

### PR DESCRIPTION
**Pull request by nuclearwu**
Signed-off-by: wuzhongjian wuzhongjian_yewu@cmss.chinamobile.com

### Summary
🆕📝🚀

Add support for specifying namespace in workflow steps. This change adds a `namespace` parameter to the `apply-component` workflow step definition and logic in various files, and updates the documentation accordingly.

> _`apply-component`_
> _now accepts `namespace`_
> _a spring cleaning cue_

### Walkthrough
*  Add `namespace` parameter to `apply-component` workflow step definition and handler ([link](https://github.com/kubevela/kubevela/pull/6228/files?diff=unified&w=0#diff-d04ff4062f6d641448d51084a8340da502252bad348799cd4f8d89f5fdbbdfa7R22-R23), [link](https://github.com/kubevela/kubevela/pull/6228/files?diff=unified&w=0#diff-4170e9b730a11fdd789c7b33a0871e4a67c4e37910815cc262286ac8ec076212R255), [link](https://github.com/kubevela/kubevela/pull/6228/files?diff=unified&w=0#diff-4170e9b730a11fdd789c7b33a0871e4a67c4e37910815cc262286ac8ec076212L291-R294), [link](https://github.com/kubevela/kubevela/pull/6228/files?diff=unified&w=0#diff-b294236ceeaba10f0beef13117ca02e3063a2c33a6f7c67ae0f2195e865fa9e2L8-R10), [link](https://github.com/kubevela/kubevela/pull/6228/files?diff=unified&w=0#diff-b294236ceeaba10f0beef13117ca02e3063a2c33a6f7c67ae0f2195e865fa9e2L21-R23), [link](https://github.com/kubevela/kubevela/pull/6228/files?diff=unified&w=0#diff-17fef7796b683f2995e4bd6650af842e7d1c0c116471d2f206ec405b5c192256R17-R18))
* Update example and documentation of using the `apply-component` workflow step ([link](https://github.com/kubevela/kubevela/pull/6228/files?diff=unified&w=0#diff-9e546a8bf33fb0f29a44924f3621c3016273cc56d76682fd38be99e39f1bca1aL27-R28))

Fixes https://github.com/kubevela/kubevela/issues/6223

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Comment by nuclearwu** [comment]
@wonderflow @Somefive please review

File: pkg/controller/core.oam.dev/v1beta1/application/generator.go [hunk 1]
```diff
@@ -252,6 +252,7 @@ func convertStepProperties(step *workflowv1alpha1.WorkflowStep, app *v1beta1.App
 	o := struct {
 		Component string `json:"component"`
 		Cluster   string `json:"cluster"`
+		Namespace string `json:"namespace"`
```
Comment (wonderflow) [inline pkg/controller/core.oam.dev/v1beta1/application/generator.go:255]: please use the application namespace if empty.

Comment (nuclearwu) [reply pkg/controller/core.oam.dev/v1beta1/application/generator.go:255]: have done @wonderflow

**Review by FogDong — APPROVED** [review]

**Comment by nuclearwu** [comment]
ping @wonderflow

**Review by wonderflow — APPROVED** [review]

**Pull request merged**